onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -123,6 +123,7 @@ Status Conv<T>::UpdateState(OpKernelContext* context, bool bias_expected) const
     if (w_dims_changed) {
       s_.last_w_dims = w_dims;
       s_.cached_benchmark_results.clear();
+      ORT_RETURN_IF_ERROR(s_.w_desc.Set(w_dims, CudnnTensor::GetDataType<CudaT>()));
```

Comment:
LGTM. As synced offline, please consider adding a test with a single node Conv model with the first Run()'s input being empty followed by a Run() that that actually has data to process.

Reply:
Btw: I think this should come after the `rank < 2` logic handling because `w_dims` is padded in that case. That is causing the test failures as well.

Reply:
Indeed, I moved it later.